src/libAtomVM/bif.h

```diff
@@ -39,11 +39,20 @@ extern "C" {
 
 #define MAX_BIF_NAME_LEN 260
 
+struct SubBinarySlice
+{
+    avm_int_t pos;
+    avm_int_t len;
+    size_t heap_size;
+};
+
```

Comment:
The alternative is to wrap everything in one function that does GC and takes roots to preserve.